source/blog/2019-03-02-update-on-module-unification-and-octane.md

```diff
@@ -0,0 +1,427 @@
+may have been a blessing in disguise. It gave us time to implement angle
+bracket syntax for components, which allowed template imports to seem
+feasible again—which meant that we now had a solution that seemed to address
+both the scoped package problem and the ergonomics problem.
+
+I believe the dead-end we found ourselves in was a sign from the universe
+that something better was just around the corner. Time will tell, but my
+hunch is that template imports solve some important problems much more elegantly
+than what we had before.
+
+This process also pushed us to explore single-file components, which I think
+will be a surprisingly big productivity improvement for Ember developers.
+
+### What's Next
+
+Hopefully, this post helps provide some context about the state of MU. Of
+course, what I've written here is my personal, imperfect recollection of
+events, simplified for clarity. The reality of technical design is messy and
+feels a lot more like going around in circles than the tidy sequence I've
+presented here.
+
+I will also mention that, as a project, I think we've dramatically improved
+how we design, implement, validate, and iterate on features since we
+originally started the Module Unification RFC. The MU RFC is the last of the
+proposals from the "mega-RFC" era, where we had a tendency to do a ton of
+upfront design and implementation before we had any feedback from real users.
+
+Nowadays, I think we're a lot better about making sure RFCs are relatively
+small and focused on doing one thing and doing it well. We also tend to
+prioritize exposing hooks or other primitives that let us test out new ideas
+in addons. This allows us to improve designs based on early feedback,
+without the strict stability requirements that come with landing something in
+the framework proper.
+
+This has worked well for things like `ember-decorators` and
+`@glimmer/component`, where real world feedback and the ability to make
+breaking changes based on that feedback was critical. I'm hopeful that a
+similar strategy for template imports will be just as successful.
+
+I recently posted the [SFC & Template Import Primitives
+RFC](https://github.com/emberjs/rfcs/pull/454), which is the first of several
+RFCs describing changes to MU or new features that complement MU. You should
+see more RFCs related to this topic in the coming weeks.
+
+My sincere thanks to everyone who has worked so hard on MU, template imports,
+and other related efforts. I'm proud to be part of a community that refuses
+to charge ahead with something we know isn't right. Ember's longevity is, at
+least in part, explained by our willingness to make these kinds of hard
+decisions.
+
+I'm so excited about Ember, our roadmap, and the newfound energy in our
+community. In 2019, a thriving Ember is more important for the web than ever.
+Thank you for being a part of our community, and I hope to see you at
+EmberConf in a few weeks. It's gonna be a good one.
```

Comment:
[EmberConf](https://emberconf.com) in a few weeks. It's gonna be a good one.

Reply:
```suggestion
[EmberConf](https://emberconf.com) in a few weeks. It's gonna be a good one.
```